tooling/karaf-maven-plugin/src/main/java/org/apache/karaf/tooling/KarMojo.java

```diff
@@ -209,6 +210,13 @@ private File resolveFile(String file) {
                         resources.add(resourceToArtifact(bundle.getLocation(), false));
                     }
                 }
+                for (Conditional conditional : feature.getConditional()) {
+                    for (BundleInfo bundle : conditional.getBundles()) {
+                        if (ignoreDependencyFlag || (!ignoreDependencyFlag && !bundle.isDependency())) {
```

Comment:
```suggestion
                        if (ignoreDependencyFlag || !bundle.isDependency()) {
```

If the RHS of the `||` is evaluated at all, `!ignoreDependencyFlag` is always `true`.

Reply:
I have simplified the conditions.
Wouldn't really be comfortable adding integration tests, being this my first contribution to the project.